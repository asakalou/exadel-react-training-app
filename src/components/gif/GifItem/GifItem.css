.gif-item {
  position: relative;
  cursor: pointer;
  margin-bottom: 10px; }
  .gif-item-fav {
    position: absolute;
    top: -15px;
    right: -20px;
    padding: 5px;
    font-size: 1.3rem;
    width: 100%;
    text-align: right; }
