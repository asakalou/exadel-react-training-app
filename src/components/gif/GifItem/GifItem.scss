.gif-item {

  position: relative;
  cursor: pointer;
  margin-bottom: 10px;

  &-fav {
    position: absolute;
    top: -15px;
    right: -20px;
    padding: 5px;
    font-size: 1.3rem;
    width: 100%;
    text-align: right;
  }

}