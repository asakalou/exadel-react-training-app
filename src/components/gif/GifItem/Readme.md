Basic example

```js
<GifItem 
    url="https://media3.giphy.com/media/tA8p2hy4KR29i/200w.gif" 
    alt="Example"
/>
```


With favourites but not favourite

```js
<GifItem 
    url="https://media3.giphy.com/media/tA8p2hy4KR29i/200w.gif" 
    alt="Example" 
    fav={true}
    onFavClick={() => alert('Favourite!')}
/>
```


With favourites and favourite

```js
<GifItem 
    url="https://media3.giphy.com/media/tA8p2hy4KR29i/200w.gif" 
    alt="Example" 
    fav={true} 
    favourite={true}
    onFavClick={() => alert('Favourite!')}
/>
```