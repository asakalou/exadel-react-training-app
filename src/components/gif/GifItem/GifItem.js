import React from 'react';
import PropTypes from 'prop-types';

import './GifItem.css';

const propTypes = {
    url: PropTypes.string.isRequired,
    alt: PropTypes.string.isRequired
};

const GifItem = ({url, alt}) => {
    return (
        <div className="gif-item">
            <img src={url} alt={alt}/>
        </div>
    )
};
GifItem.propTypes = propTypes;

export default GifItem;