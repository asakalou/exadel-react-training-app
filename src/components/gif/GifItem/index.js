import GifItem from './GifItem';

export default GifItem;