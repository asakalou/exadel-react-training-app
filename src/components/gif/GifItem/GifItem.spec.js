import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';

import GifItem from "./GifItem";

describe('<GifItem/>', () => {


    it('should render props correctly', () => {
        const gifItem = shallow(<GifItem url="gif_url" alt="alt text"/>);

        const img = gifItem.find('img');

        expect(img.prop('src')).toEqual('gif_url');
        expect(img.prop('alt')).toEqual('alt text');

    });

    it('should render snapshot correctly', () => {
        const tree = renderer
            .create(<GifItem url="url" alt="alt"/>)
            .toJSON();

        expect(tree).toMatchSnapshot();
    })


});