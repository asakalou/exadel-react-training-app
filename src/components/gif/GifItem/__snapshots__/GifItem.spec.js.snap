// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifItem/> should render snapshot correctly 1`] = `
<div
  className="gif-item"
>
  <img
    alt="alt"
    src="url"
  />
</div>
`;
